src/main/java/com/github/alex1304/jdash/client/GDLevelDemonRatingRequest.java

```diff
@@ -0,0 +1,54 @@
+package com.github.alex1304.jdash.client;
+
+import com.github.alex1304.jdash.entity.DemonDifficulty;
+import com.github.alex1304.jdash.exception.GDClientException;
+import com.github.alex1304.jdash.util.Routes;
+
+import java.util.Map;
+import java.util.Objects;
+
+class GDLevelDemonRatingRequest extends AbstractAuthenticatedGDRequest<Void> {
+
+    private final long levelID;
+    private final DemonDifficulty difficulty;
+
+    GDLevelDemonRatingRequest(AuthenticatedGDClient client, long levelID, DemonDifficulty difficulty) {
+        super(client);
+        this.levelID = levelID;
+        this.difficulty = Objects.requireNonNull(difficulty);
+    }
+
+    @Override
+    public String getPath() {
+        return Routes.RATE_LEVEL_DEMON;
+    }
+
+    @Override
+    void putParams(Map<String, String> params) {
+        params.put("levelID", ""+levelID);
+        params.put("rating", ""+(difficulty.ordinal()+1));
+        params.put("secret", "Wmfp3879gc3"); //Overrides the default one
+    }
+
+    @Override
+    Void parseResponse0(String response) throws GDClientException {
+        if (!response.equals("1")) {
+            throw new IllegalArgumentException("Unknown response: " + response);
```

Comment:
Create a new subclass of `GDClientException` in the `com.github.alex1304.jdash.exception` package.

Reply:
well you're supposed to use them here in place of your IllegalArgumentExceptions

Reply:
i think i'm crazy.. b5d278e